lxd/storage/drivers/utils_lvm.go

```diff
@@ -0,0 +1,262 @@
+			exitError, ok := runErr.Err.(*exec.ExitError)
+			if ok {
+				waitStatus := exitError.Sys().(syscall.WaitStatus)
+				if waitStatus.ExitStatus() == 5 {
+					// volume group not found
+					return false, nil
+				}
+			}
+		}
+
+		return false, fmt.Errorf("error checking for volume group \"%s\"", vgName)
+	}
+
+	return true, nil
+}
+
+// LVMGetLVCount gets the count of volumes in a volume group.
+func LVMGetLVCount(vgName string) (int, error) {
+	output, err := shared.TryRunCommand("vgs", "--noheadings", "-o", "lv_count", vgName)
+	if err != nil {
+		return -1, err
+	}
+
+	output = strings.TrimSpace(output)
+	return strconv.Atoi(output)
+}
+
+// LVMThinpoolExists checks whether the specified thinpool exists in a volume group.
+func LVMThinpoolExists(vgName string, poolName string) (bool, error) {
+	output, err := shared.RunCommand("vgs", "--noheadings", "-o", "lv_attr", fmt.Sprintf("%s/%s", vgName, poolName))
+	if err != nil {
+		runErr, ok := err.(shared.RunError)
+		if ok {
+			exitError, ok := runErr.Err.(*exec.ExitError)
+			if ok {
+				waitStatus := exitError.Sys().(syscall.WaitStatus)
+				if waitStatus.ExitStatus() == 5 {
+					// pool LV was not found
+					return false, nil
+				}
+			}
+		}
+
+		return false, fmt.Errorf("error checking for pool \"%s\"", poolName)
+	}
+	// Found LV named poolname, check type:
+	attrs := strings.TrimSpace(string(output[:]))
+	if strings.HasPrefix(attrs, "t") {
+		return true, nil
+	}
+
+	return false, fmt.Errorf("pool named \"%s\" exists but is not a thin pool", poolName)
+}
+
+// LVMDevPath returns the path to the LVM volume device.
+func LVMDevPath(projectName, lvmPool string, volumeDBTypeName string, lvmVolume string) string {
+	lvmVolume = project.Prefix(projectName, lvmVolume)
+	if volumeDBTypeName == "" {
+		return fmt.Sprintf("/dev/%s/%s", lvmPool, lvmVolume)
+	}
+
+	return fmt.Sprintf("/dev/%s/%s_%s", lvmPool, volumeDBTypeName, lvmVolume)
+}
+
+// LVMVolumeExists checks whether the specified logical volume exists.
+func LVMVolumeExists(lvDevPath string) (bool, error) {
+	_, err := shared.RunCommand("lvs", "--noheadings", "-o", "lv_attr", lvDevPath)
+	if err != nil {
+		runErr, ok := err.(shared.RunError)
+		if ok {
+			exitError, ok := runErr.Err.(*exec.ExitError)
+			if ok {
+				waitStatus := exitError.Sys().(syscall.WaitStatus)
+				if waitStatus.ExitStatus() == 5 {
+					// logical volume not found.
+					return false, nil
+				}
+			}
+		}
+
+		return false, fmt.Errorf("error checking for logical volume \"%s\"", lvDevPath)
+	}
+
+	return true, nil
+}
+
+// LVMCreateThinpool creates a thin pool logical volume.
+func LVMCreateThinpool(lvmVersion string, vgName string, thinPoolName string) error {
+	exists, err := LVMThinpoolExists(vgName, thinPoolName)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		return nil
+	}
+
+	err = lvmCreateDefaultThinPool(lvmVersion, vgName, thinPoolName)
+	if err != nil {
+		return err
+	}
+
+	poolExists, err := LVMThinpoolExists(vgName, thinPoolName)
```

Comment:
@stgraber I wanted to draw your attention to this function, as I've changed its signature to remove the need for `*state.State` and the `lvFsType`.

The old function is here: https://github.com/lxc/lxd/pull/6636/files#diff-78f58b312bc5d2547c9d3b45fab90eecL884

The `lvFsType` argument was not used, makes sense, because thin pools dont have a filesystem.

The `*state.State` argument was used after thin pool creation to call `storageLVMGetThinPoolUsers` which seemingly is more suited to checking usage of an existing thin pool and preventing renames: https://github.com/lxc/lxd/blob/6b1953776b1d73ea3c878dbe1e928e8e64378fd0/lxd/storage_lvm_utils.go#L724

Instead I just replaced it with a call to LVMThinpoolExists to check the thinpool was created OK (this seems unnecessary too as the previous create command succeeded, but presume there is a historical reason this is preferred).